### PR TITLE
Refactor PokeApp & components, move fetching to child components, lazyload.

### LINKED ISSUE #45: Refactor modal, search bar, and other components to decrease file size, and add loading spinner
**Is your feature request related to a problem? Please describe.**
In PokeList render, the ternary is nested. There are many parts of large components that could be broken down and move to  separate files.

**Describe the solution you'd like**
Refactor and move h1 and individual PokeCard to its own component to improve readability, and other components to their own files.
Add spinner during fetch.

### LINKED ISSUE #68: Need a way to handle non-Eng flavor text
## Expected Behavior
Should be showing English flavor text in first tab of Modal..

## Actual/Current Behavior
Not showing English flavor text 
Screen Shot 2021-07-30 at 10 56 54 AM

API has English flavor text.
Screen Shot 2021-07-30 at 11 00 27 AM

## Steps to Reproduce the Problem
  1. Search pokemon with Dex. No. greater than/or equal to `808`.
  2. 
  3. 

### If you are not a member of this project, _skip this step_

_How do the users test this change?_

  1. Search for Pokemon called `necrozma`.
  2. Open modal and see first tab. 
  3. Observe.

## Environment
MacOS, latest version of Chrome browser.

### LINKED ISSUE #69: More modal issues - Empty Moves array
## Expected Behavior
1. Should show English flavor text on first render, and after switching tabs.
2. Evolution tab should show sprite for  pokemon`zacian-hero`.

## Actual/Current Behavior
1. Showing non-Japanese text in first tab.
2. Getting multiple errors, from EvolutionTab and Moves when switching between tabs.

## Steps to Reproduce the Problem
1. Search `zacian-hero`.
2. Open modal. Switch between tabs.
3. Observe.

## Pre-Testing TODOs

_What needs to be done before testing._
1. Fix/add conditional rendering?

### If you are not a member of this project, _skip this step_

_How do the users test this change?_
1. Search `zacian-hero`.
2. Open modal. Switch between tabs.
3. Observe.

## Environment
MacOS, latest Chrome version.

Screen Shot 2021-07-30 at 11 14 52 AM

Screen Shot 2021-07-30 at 11 14 44 AM

Screen Shot 2021-07-30 at 11 27 51 AM

Screen Shot 2021-07-30 at 11 34 03 AM

### LINKED ISSUE #71: Modal tabs browser console error
## Expected Behavior
No errors in browser console.

## Actual/Current Behavior
Modal is functioning fine, but there is an error in browser related to the material ui tabs.

## Steps to Reproduce the Problem
  1. `npm start`
  2. Open modal.
  3. Check browser console.

## Pre-Testing TODOs
  1. Modify code in TabBar, and potentially PokemonDetails(About), and either use PropTypes or don't.
  2. Maybe consider switching to different modal. 

### If you are not a member of this project, _skip this step_

_How do the users test this change?_

  1. Open modal.
  2. Check browser console.
  3. Observe if there are any DOM nesting error.
Screen Shot 2021-07-30 at 2 30 23 PM

### LINKED ISSUE #73: Too many fetch calls at start of app load page
## Expected Behavior

Pokeapp should only be fetching to the first 20 pokemon, this is indicated in the console where it shows all the GET actions

## Actual/Current Behavior

_Describe what is currently happening._

Displays GET for first 20 pokemon, then for pokemon-species, and pokmon-moves.

## Steps to Reproduce the Problem

  1. Open browser console
  2. Load or refresh pokeapp

image

image

### LINKED ISSUE #82: Abilities are not displaying correct values for varying pokemon
## Expected Behavior

Pokemon abilities are displayed in the about tab of the selected pokemon

## Actual/Current Behavior

Abilities under the about tab are only rendering "Overgrow, Chlorophyll" and not the correct ability for the selected pokemon unless the user searches.

## Testing Steps

_How do the users test this change?_

  1. Click a pokemon card to open modal
  2. Observe value displayed under abilities

### LINKED ISSUE #85: Code refactoring
**Is your feature request related to a problem? Please describe.**
There aren't enough comments, rename variables, and functions to optimize.

### LINKED ISSUE #87: Stats graph does not reflect Pokemon stats correctly.
## Expected Behavior
Pokemon stats graph in modal should match the pokemon stats.

## Actual/Current Behavior
Example:
Screen Shot 2021-08-05 at 4 36 07 PM
Screen Shot 2021-08-05 at 4 36 23 PM
 Pikachu speed stat should be highest stat.

## Steps to Reproduce the Problem
  1. Open app.
  2. Click card to open modal.
  3. Go to Base Stats tab and observe.

### If you are not a member of this project, _skip this step_

  1. Navigate to x screen
  2. Click on y button
  3. Observe z result

## Environment

_Please provide **OS Version** and **Platform/Browser**._
Chrome on MacOS.

### PR DESCRIPTION
**Pull request by connielion**
## Changes
1. Installed 'react-finite-scroll-component' and wrapped it around list of PokeCards.
2. Added spinner(Loader); spinner styling imported into `main.scss` . 
3. Moved TabPanel out of Tabar to its own file.
4. Created Searched component and moved part of conditional rendering there from PokeList.
5. Moved modal data fetching from top-level component down to PokeCard and passed it down to descendants.
6. Defined `findDescriptionByLanguage`, which takes the following 3 arguments: 1) `array`(either `genera`, `flavor_text_entries`), 2) `targetKey`('genus' or 'flavor_text'), 3) `lang`(`en` for English as the target language); makes code more DRY. Invoke it in `triggerModalData`.
7. Moved `getPokemonModalAboutContent` definition to PokeCard, and invocation into `triggerModalData`.
8. Moved `triggerModalData` and `setModalIsOpen` calls into new `prepareModal` function.
9. Add state hooks `generaString` and `flavorText` in PokeCard to be passed down to the tabs as `genera` and `flavorText`.
10. Passed `dexNo` down to Moves component, imported `fetchMove` and `fetchPokemon` from `util`.
11. Moved `getMovesByPokemon` to Moves component, invoke it in `useEffect`, create hook for setting array of moves objects to be mapped in `moveInfo`.
12. Add `abilities` and `stats` into destructuring from `pokeObj` in `generateSortedCards` to fix #82 and #87.
13. Fixed minor scss error.
14. Fixed key props errors in EvolutionsTab and Moves tab.

## Purpose
1. Refactor and added some comments.
2. Decrease prop drilling.
3. Optimization and decrease file size.
4. Make code more DRY.

## Approach
- Refactor fetching and move them to only the child components where the data is needed instead of in top-level component.

- Define a more DRY function that finds English descriptions and genera since the `flavor_text_entries` and `genera` array are structured the same way.

Closes #45 
Closes #68 
Closes #69 
Closes #71 
Closes #73 
Closes #82 
Closes #85 
Closes #87

### CODE REVIEW

File: src/components/Loader.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <div className="lds-roller">
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+    </div>
```
Comment (Lap343) [inline src/components/Loader.js:14]: hmmm. Curious what this is for

File: src/components/PokeApp.js [hunk 2]
```diff
@@ -17,8 +17,8 @@ function PokeApp() {
   const [allPokemons, setAllPokemon] = useState([]);
   // Hook below is used for conditional rendering 1 Pokemon card user searched, and search on keypress
   const [userDidSearch, setUserDidSearch] = useState(false);
-  const [modalData, setModalData] = useState([]);
 
+  const [isLoading, setIsLoading] = useState(true);
```
Comment (Lap343) [inline src/components/PokeApp.js:21]: nice having a loading state so the app will wait for the promise to finish

**Review by Lap343 — APPROVED** [review]
Connie the issue master. Nice refactor and getting all those issues done for

File: src/components/Loader.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+  return (
+    <div className="lds-roller">
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
+      <div></div>
```
Comment (BlackCubes) [inline src/components/Loader.js:13]: Empty 👀

File: src/components/Moves.js [hunk 4]
```diff
@@ -22,7 +22,35 @@ import {
...
+    const nMoves = [];
+    for (let i = 0; i < n; i += 1) {
+      const moveUrl = movesArray[i].move.url;
+      try {
+        const moveData = await fetchMove(moveUrl);
+        if (moveData) nMoves.push(moveData);
+      } catch (error) {
+        console.error(error);
+      }
+    }
```
Comment (BlackCubes) [inline src/components/Moves.js:38]: Couldn't you use `Promise.all` to fetch all the moves URL?

**Review by BlackCubes — APPROVED** [review]
Jesus........ the PR Master. This was really good! Beautiful

File: src/components/Moves.js (hunk 4 above)
Comment (kmausolf) [reply src/components/Moves.js:38]: The thing with `Promise.all` is that if even one promise rejects, it will automatically reject.  This is a bit safer.

Comment (BlackCubes) [reply src/components/Moves.js:38]: Makes sense

**Pull request merged**